content/en/posts/mesa-optimizers.md

```diff
@@ -0,0 +1,40 @@
+
+Does this mean Bostrom and Yudkowsky were misguided in their worries, and trained AIs will be harmless, since they themselves are not optimizers? Unfortulately not, as this is where mesa optimizers enter the playing field. "Mesa" is simply the opposite of "meta", so something like "one level down". The idea is that a training process (in this context usually called the *base optimizer*) could potentially select for programs that themselves are optimizers (the *mesa optimizer*). Theoretically this makes sense: an agent that is actively optimizing for some training target has a good chance of being among the best agents when it comes to actually achieving said training target. Whereas non-optimizer agents likely perform worse in comparison.
+
+However, the training loop is not some omniscient or omnipotent entity that just happens to find the best possible program or agent. It is still a relatively simple process, that usually starts with some randomized initialization of model weights, and then performs a greedy search process (such as SGD), iteratively improving the model weights. It's unclear under which circumstances exactly such a process would actually *find* a set of model weights that would yield a mesa optimizer for any given initialization.
+
+Yet, despite these uncertainties, there are at least two reasons to take this idea quite seriously:
+
+1. Even if currently training loops mostly don't end up creating mesa optimizers and maybe even lack the capabilities to identify such programs in the vast multi-dimensional search space of model weights, we just need to look one level up: behind each ML training process, there's a bunch of **human researchers** who constantly tweak the training regime in order to come up with the most capable possible models. These humans **have an incentive to keep adjusting the training process until it will eventually be able to create mesa optimizers**.
+2. **Humans could be considered exemplary for mesa optimization**: a simple outside process, namely evolution, has run an optimization on living beings, selecting inclusive genetic fitness, and eventually (quite misaligned, from the point of view of evolution) optimizers emerged in the form of humans.
+
+One caveat to all of this is that the concept of an "optimizer" is difficult to define cleanly. And while it's easy to think of optimizer-ness as a binary property, it's probably more continuous, with some AIs being more and others less "optimizer-like". Looking at the example of humans mentioned above, it's questionable to what degree humans really are optimizers. Some individuals are optimizing for their goals much more than others, but practically all humans are investing some effort to shape the world in accordance with their preferences. So while many humans exhibit optimizer-like behaviour, we're certainly far from *perfect* optimizers.
```

Comment:
If humans are not optimizer I wonder what IS an optimizer? Do you have a source for this paragraph or is it your own idea? That does not make a difference (an argument should stand on its own not rest on authority) but I would be interested to read more about this line of reasoning if there is a source.

Reply:
I don't really have a source for this, but the point I'm trying to make also seems pretty robust to me in itself, so maybe I'm just not getting it across properly. It's basically that "being an optimizer" is probably best thought of as a spectrum, and humans fall somewhere on that, with a relatively wide distribution. Almost all are ">0", but also almost all humans are far from "pure optimizers". E.g. humans seem very different in their optimization behaviour from say your typical paperclip maximizer. To motivate this a bit further:

- most humans don't have any explicit representation of a goal function they're optimizing for
- most humans have strong temporal variance in their goals
- to the degree that humans are optimizing for something, the vast majority probably invests only a minority of their energy output into that topic

But maybe all of this is too debatable to put into such an article? I personally would think that the claim "humans are optimizers" would be more controversial though. :)

Reply:
Changed part of this to `Looking at the example of humans mentioned above, it's not perfectly clear where on this optimizer continuum most humans would fall.` now, whic maybe goes a little less in the "humans are not optimizers" direction.